react/Filename/index.jsx

```diff
@@ -35,11 +35,35 @@ const Filename = ({ icon, filename, extension }) => {
 
 Filename.propTypes = {
   /** Filename icon */
-  icon: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+  icon: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.object,
+    PropTypes.func
+  ]),
   /** folder or file name */
   filename: PropTypes.string.isRequired,
   /** If a file name, you can specify the extension */
-  extension: PropTypes.string
+  extension: PropTypes.string,
+  variant: PropTypes.oneOf([
+    'body1',
+    'body2',
+    'button',
+    'caption',
+    'h1',
+    'h2',
+    'h3',
+    'h4',
+    'h5',
+    'h6',
+    'inherit',
+    'overline',
+    'subtitle1',
+    'subtitle2'
+  ])
```

Comment:
Not a big fan of duplicating the props. 

Can't we use that instead https://github.com/mui/material-ui/blob/v4.x/packages/material-ui/src/Typography/Typography.js#L164 ?